### ISSUE TITLE
FR: allow to disable auto-stop on running workspaces

### ISSUE THREAD

**Issue by janLo** [coder/coder#9775]
I have some templates with a 24h auto-stop-timeout, but where the user can disable the autostop feature altogether if needed. This works well as most of the users are fine with the auto-stop, and we can save resources.

But if someone needs to preserve a state e.g. over the weekend, it is not possible to disable the auto-stop for a running workspace. If the user tries to toggle the auto-stop toggle, a dialog pops up and asks if the workspace should restart now or if the setting should be applied later at the next restart. This is not what the user wanted in this case.

I don't understand why the restart is required, and I think, it would be a great UX improvement if the user could disable the autostop without having to restart (stop) the running workspace.

**Comment by bpmct** [comment]
This makes a ton of sense, I'll see if we can prioritize it next month

**Comment by stirby** [comment]
If your users are able to customize the Autostop timer on their workspaces, they could use the [`schedule override-stop`](https://coder.com/docs/v2/latest/cli/schedule_override-stop) command.

You should be able to use `coder schedule override-stop my-workspace 72h` to keep the workspaces alive over the weekend. However, this command doesn't support truly indefinite Autostop suspension (I believe the max is 132 hours).

Would this remedy your issue? 

We still may add some functionality to change Autostop without restarting the workspace down the road.

**Comment by mambon2** [comment]
+1

**Comment by DanielleMaywood** [comment]
As it's been a while since this feature was requested I'm going to quickly mention what can be done _currently_ to achieve similar to this goal whilst I work on the design for this 👍. 

You can use [`coder schedule extend <workspace-name> <duration from now>`](https://coder.com/docs/reference/cli/schedule_extend) (previously `override-stop`) to extend the deadline of a running container. There is no upper limit to the `<duration from now>` argument, meaning you could pass something like `8736h` to give the workspace an entire year before it shuts down. Now this isn't the same as disabling autostop but it should be sufficient for your use case for now.

**Comment by mambon2** [comment]
Yes this issue has been addressed.  However, making changes to workspace schedule still requires a restart.  Not sure how easy it would be to make those changes without a restart.
